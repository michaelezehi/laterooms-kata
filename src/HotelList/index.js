import React, { Component, Fragment } from 'react';
import { ListGroup, ListGroupItem, Label } from 'react-bootstrap';

const HotelList = ({ listHotels }) => {

  const hotelItems = listHotels.map((h, i) => (
    <ListGroupItem key={i} header={`${h.Name} - ${h.StarRating} stars`}>
      { h.Facilities.map((f, ind) => (
        <Fragment key={ind}>
          <Label>{ f }</Label>{ ' ' }
        </Fragment>
      )) }
    </ListGroupItem>
  ));

  return (
    <ListGroup>
      { hotelItems }
    </ListGroup>
  )
};

export default HotelList;
