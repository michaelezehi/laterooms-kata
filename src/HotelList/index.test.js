import React from 'react';
import renderer from 'react-test-renderer';
import HotelList from './index';

import { HOTELS } from '../hotelData';

it('renders without crashing', () => {
  const component = renderer.create(
    <HotelList listHotels={HOTELS} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
