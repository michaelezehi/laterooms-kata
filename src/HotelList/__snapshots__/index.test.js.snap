// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    <h4
      className="list-group-item-heading"
    >
      hotelone - 5 stars
    </h4>
    <p
      className="list-group-item-text"
    >
      <span
        className="label label-default"
      >
        car park
      </span>
       
      <span
        className="label label-default"
      >
        pool
      </span>
       
    </p>
  </li>
  <li
    className="list-group-item"
  >
    <h4
      className="list-group-item-heading"
    >
      hoteltwo - 3 stars
    </h4>
    <p
      className="list-group-item-text"
    >
      <span
        className="label label-default"
      >
        car park
      </span>
       
      <span
        className="label label-default"
      >
        gym
      </span>
       
    </p>
  </li>
  <li
    className="list-group-item"
  >
    <h4
      className="list-group-item-heading"
    >
      hotelthree - 3 stars
    </h4>
    <p
      className="list-group-item-text"
    />
  </li>
</ul>
`;
